### PR TITLE
[GPU] Minor fixes for regression from runtime buffer fusing for crop

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Minor changes to fix regression issue from https://github.com/openvinotoolkit/openvino/pull/25060

### Tickets:
 - 137277

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp [hunk 1]
```diff
@@ -479,6 +479,12 @@ bool crop_in_place_optimization::match(const program_node& node,
         // TODO: Need to allow optimization for gemm user
         if (node.is_dynamic() && (user->is_type<convolution>() || user->is_type<gemm>()))
             return false;
+        // If the user is strieded slice and the output shape is already static,
+        // the padding w/ dyn pad mask will not be propagated properly at runtime
+        if (user->is_type<strided_slice>()) {
+            if (node.is_dynamic() && !user->get_output_layout().is_dynamic())
+                return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:486]: But if the user's output is static, isnt there same issue regardless it is strided slice or not?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp]: Agree. It doesn't seem to matter whether the user node is a specific type. The condition have been made more general.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp [hunk 2]
```diff
@@ -468,6 +468,10 @@ bool crop_in_place_optimization::match(const program_node& node,
 
     const auto& crop_layout = crop_params.get_output_layout();
     for (auto user : node.get_users()) {
+        // If the user node's output shape is already static, the padding
+        // w/ dyn pad mask will not be propagated properly at runtime
+        if (node.is_dynamic() && !user->get_output_layout().is_dynamic())
+            return false;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp:474]: I think even the first condition is not needed :)

**Review by kelvinchoi-intel — APPROVED** [review]
LGTM

**Review by yeonbok — APPROVED** [review]

**Pull request merged**